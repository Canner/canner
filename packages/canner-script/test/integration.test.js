/** @jsx builder */
import builder, {Block} from '../src';
import compiler from 'canner-compiler';
import serializer from './serializer';
expect.addSnapshotSerializer(serializer);
const {Traverser, Parser} = compiler;
function compile(schema, visitors) {
  const parser = new Parser();
  const tree = parser.parse(schema);
  const traverser = new Traverser(tree);
  visitors.forEach(visitor => {
    traverser.addVisitor(visitor);
  });
  const componentTree = traverser.traverse();
  return componentTree;
}

describe('integration with qa-compiler, test object', () => {
  it(`integration 1`, () => {
    const {schema, visitors} = <root>
      <object keyName="info">
        <Block>
          <string keyName="name"/>
        </Block>
      </object>
    </root>;
    expect(compile(schema, visitors)).toMatchSnapshot();
  });

  it(`integration 2`, () => {
    const {schema, visitors} = <root>
      <object keyName="info">
        <Block>
          <string keyName="name"/>
        </Block>
        <string keyName="nickname"/>
      </object>
    </root>;
    expect(compile(schema, visitors)).toMatchSnapshot();
  });

  it(`integration 3`, () => {
    const {schema, visitors} = <root>
      <object keyName="info">
        <Block>
          <string keyName="name"/>
          <string keyName="nickname"/>
        </Block>
      </object>
    </root>;
    expect(compile(schema, visitors)).toMatchSnapshot();
  });

  it(`integration 4`, () => {
    const {schema, visitors} = <root>
      <object keyName="info">
        <Block>
          <object keyName="info1">
            <string keyName="name"/>
          </object>
        </Block>
      </object>
    </root>;
    expect(compile(schema, visitors)).toMatchSnapshot();
  });

  it(`integration 5`, () => {
    const {schema, visitors} = <root>
      <object keyName="info">
        <Block>
          <object keyName="info1">
            <Block>
              <string keyName="name"/>
            </Block>
          </object>
        </Block>
      </object>
    </root>;
    expect(compile(schema, visitors)).toMatchSnapshot();
  });

  it(`
  integration 6`, () => {
    const {schema, visitors} = <root>
      <object keyName="info">
        <object keyName="info1">
          <Block>
            <object keyName="info2">
              <string keyName="name3"/>
            </object>
          </Block>
        </object>
      </object>
      
    </root>;
    expect(compile(schema, visitors)).toMatchSnapshot();
  });

  it(`integration 7`, () => {
    const {schema, visitors} = <root>
      <object keyName="info" title="Info">
        <Block>
          <string keyName="name" title="Name" />
          <string keyName="description" title="Desc" />
        </Block>
        <object keyName="info" title="Info">
          <object keyName="info2" title="Info2">
            <string keyName="name2" title="Name" />
          </object>
        </object>

      </object>
    </root>;
    expect(compile(schema, visitors)).toMatchSnapshot();
  });

  it(`integration 8`, () => {
    const {schema, visitors} = <root>
      <object keyName="info" title="Info">
        <Block>
          <string keyName="name" title="Name" />
          <string keyName="description" title="Desc" />
        </Block>
        <object keyName="info" title="Info">
          <object keyName="info2" title="Info2">
            <Block>
              <string keyName="name2" title="Name" />
            </Block>
          </object>
        </object>
      </object>
      <array keyName="info3" title="Info3">
        <Block>
          <string keyName="name" title="Name" />
          <string keyName="description" title="Desc" />
        </Block>
        <object keyName="info" title="Info">
          <object keyName="info2" title="Info2">
            <string keyName="name2" title="Name" />
          </object>
        </object>
      </array>
    </root>;
    expect(compile(schema, visitors)).toMatchSnapshot();
  });

  it(`integration 9`, () => {
    const {schema, visitors} = <root>
      <object keyName="info">
        <Block>
          <Block>
            <string keyName="name"/>
          </Block>
        </Block>
      </object>
    </root>;
    expect(compile(schema, visitors)).toMatchSnapshot();
  });
});

describe('array', () => {
  it('integration 1', () => {
    const {schema, visitors} = <root>
      <array keyName="posts">
        <object keyName="status">
          <Block>
            <boolean keyName="isDraft" />
          </Block>
        </object>
      </array>
    </root>;
    expect(compile(schema, visitors)).toMatchSnapshot();
  })

  it('integration 2', () => {
    const {schema, visitors} = <root>
      <array keyName="posts">
        <Block>
          <string keyName="content" />
          <object keyName="buttons">
            <boolean keyName="isDraft" />
          </object>
          <object keyName="otherinfo">
            <string keyName="address" />
            <string keyName="introduction" />
          </object>
        </Block>
      </array>
    </root>;
    expect(compile(schema, visitors)).toMatchSnapshot();
  })
})


describe('complex schema', () => {
  it('integration 1', () => {
    const {schema, visitors} = <root>
      <array keyName="posts" title="Posts">
        <object keyName="status">
          <dateTime keyName="createAt" />
          <number keyName="draft" />
        </object>
        <string keyName="featureImage" />
        <object keyName="pageProperty">
          <number keyName="onTop" />
          <number keyName="order" />
        </object>
        <object keyName="share">
          <number keyName="showShareBottom" />
          <number keyName="showGoodBottom" />
        </object>
        <object keyName="other">
          <string keyName="introduction" />
          <geoPoint keyName="position" />
        </object>
      </array>
      <object keyName="info" title="Info">
        <Block>
          <string keyName="name" title="Name" />
          <string keyName="description" title="Desc" />
        </Block>
        <object keyName="info" title="Info">
          <object keyName="info2" title="Info2">
            <string keyName="name2" title="Name" />
          </object>
        </object>
      </object>
      <array keyName="popup" title="Popup">
        <string keyName="name" title="Name" />
      </array>
      <array ui="tab" keyName="tab" title="Tab">
        <string keyName="name" title="Name" />
      </array>
      <array
        title="Title"
        uiParams={{
          columns: [{
            title: 'name',
            dataIndex: 'name',
            key: 'name'
          }]
        }}
      >
        <string keyName="name" title="Name" />
      </array>
    </root>;
    expect(compile(schema, visitors)).toMatchSnapshot();
  });

  it('integration 2', () => {
    var contactInfo = [
      {
        key: 'phones',
        options: [{
          text: 'Home',
          value: 'HOME'
        }, {
          text: 'Company',
          value: 'COMPANY'
        }, {
          text: 'Mobile',
          value: 'MOBILE'
        }]
      },
      {
        key: 'addresses',
        options: [{
          text: 'Home',
          value: 'HOME'
        }, {
          text: 'Company',
          value: 'COMPANY'
        }]
      },
      {
        key: 'emails',
        options: []
      }
    ]
    const Img = ({attributes}) => (
      <object keyName={attributes.keyName} title={attributes.title}>
        <string keyName="imageName" title="imageName" />
        <file keyName="imageSrc" title="imageSrc" contentType="images/*" />
      </object>
    );
    const {schema, visitors} = <root>
      <object keyName="info" title="MyInfo">
        <Block>
          <Block keyName="name">
          {/* <file keyName="thumbnail" title="Thumbnail" contentType="image/*" ui="image"/> */}
          <string keyName="name" title="Name"/>
          <string keyName="nickname" title="Nickname" />
          </Block>
          <string keyName="test" title="321" />
          <Img keyName="thumbnail" title="Photo" />
          <object keyName="contactInfo" title="ContactInfo">
          {
            contactInfo.map(info => (
              <array ui="table" keyName={info.key} key={info.key} uiParams={{
                columns: [{
                  title: 'Type',
                  dataIndex: 'type'
                }, {
                  title: 'Value',
                  dataIndex: 'value'
                }]
              }}>
                {
                  info && <string ui="select" 
                    uiParams={{options: info.options}}
                    keyName="type"
                  />
                }
                <string keyName="value"/>
              </array>
            ))
          }
          </object>
        </Block>
      </object>
    </root>;
    expect(compile(schema, visitors)).toMatchSnapshot();
  })
})