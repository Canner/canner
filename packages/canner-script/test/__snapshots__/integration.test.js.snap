// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`array integration 1 1`] = `
{
  "posts": {
    "nodeType": "layout.body",
    "children": [
      {
        "nodeType": "layout.block",
        "children": [
          {
            "nodeType": "component.array.table",
            "keyName": "posts",
            "hocs": "[withTitleAndDescription,withRequest,relation,miniApp,withRoute,query,routeMiniApp,connectId]",
            "children": [
              {
                "nodeType": "component.object.fieldset",
                "keyName": "status",
                "hocs": "[withTitleAndDescription,withRequest,relation,withRoute,withQuery,routeMiniApp,connectId]",
                "children": [
                  {
                    "nodeType": "layout.block",
                    "childrenName": "[isDraft]",
                    "children": [
                      {
                        "nodeType": "component.boolean.switch",
                        "keyName": "isDraft",
                        "hocs": "[withTitleAndDescription,withRequest,relation,withRoute,withQuery,routeMiniApp,connectId]"
                      }
                    ]
                  }
                ]
              }
            ]
          }
        ]
      }
    ]
  }
}
`;

exports[`array integration 2 1`] = `
{
  "posts": {
    "nodeType": "layout.body",
    "children": [
      {
        "nodeType": "layout.block",
        "children": [
          {
            "nodeType": "component.array.table",
            "keyName": "posts",
            "hocs": "[withTitleAndDescription,withRequest,relation,miniApp,withRoute,query,routeMiniApp,connectId]",
            "children": [
              {
                "nodeType": "layout.block",
                "childrenName": "[content,buttons,otherinfo]",
                "children": [
                  {
                    "nodeType": "component.string.input",
                    "keyName": "content",
                    "hocs": "[withTitleAndDescription,withRequest,relation,withRoute,withQuery,routeMiniApp,connectId]"
                  },
                  {
                    "nodeType": "component.object.fieldset",
                    "keyName": "buttons",
                    "hocs": "[withTitleAndDescription,withRequest,relation,withRoute,withQuery,routeMiniApp,connectId]",
                    "children": [
                      {
                        "nodeType": "component.boolean.switch",
                        "keyName": "isDraft",
                        "hocs": "[withTitleAndDescription,withRequest,relation,withRoute,withQuery,routeMiniApp,connectId]"
                      }
                    ]
                  },
                  {
                    "nodeType": "component.object.fieldset",
                    "keyName": "otherinfo",
                    "hocs": "[withTitleAndDescription,withRequest,relation,withRoute,withQuery,routeMiniApp,connectId]",
                    "children": [
                      {
                        "nodeType": "component.string.input",
                        "keyName": "address",
                        "hocs": "[withTitleAndDescription,withRequest,relation,withRoute,withQuery,routeMiniApp,connectId]"
                      },
                      {
                        "nodeType": "component.string.input",
                        "keyName": "introduction",
                        "hocs": "[withTitleAndDescription,withRequest,relation,withRoute,withQuery,routeMiniApp,connectId]"
                      }
                    ]
                  }
                ]
              }
            ]
          }
        ]
      }
    ]
  }
}
`;

exports[`complex schema integration 1 1`] = `
{
  "posts": {
    "nodeType": "layout.body",
    "children": [
      {
        "nodeType": "layout.block",
        "children": [
          {
            "nodeType": "component.array.table",
            "keyName": "posts",
            "hocs": "[withTitleAndDescription,withRequest,relation,miniApp,withRoute,query,routeMiniApp,connectId]",
            "children": [
              {
                "nodeType": "component.object.fieldset",
                "keyName": "status",
                "hocs": "[withTitleAndDescription,withRequest,relation,withRoute,withQuery,routeMiniApp,connectId]",
                "children": [
                  {
                    "nodeType": "component.dateTime.dateTime",
                    "keyName": "createAt",
                    "hocs": "[withTitleAndDescription,withRequest,relation,withRoute,withQuery,routeMiniApp,connectId]"
                  },
                  {
                    "nodeType": "component.number.input",
                    "keyName": "draft",
                    "hocs": "[withTitleAndDescription,withRequest,relation,withRoute,withQuery,routeMiniApp,connectId]"
                  }
                ]
              },
              {
                "nodeType": "component.string.input",
                "keyName": "featureImage",
                "hocs": "[withTitleAndDescription,withRequest,relation,withRoute,withQuery,routeMiniApp,connectId]"
              },
              {
                "nodeType": "component.object.fieldset",
                "keyName": "pageProperty",
                "hocs": "[withTitleAndDescription,withRequest,relation,withRoute,withQuery,routeMiniApp,connectId]",
                "children": [
                  {
                    "nodeType": "component.number.input",
                    "keyName": "onTop",
                    "hocs": "[withTitleAndDescription,withRequest,relation,withRoute,withQuery,routeMiniApp,connectId]"
                  },
                  {
                    "nodeType": "component.number.input",
                    "keyName": "order",
                    "hocs": "[withTitleAndDescription,withRequest,relation,withRoute,withQuery,routeMiniApp,connectId]"
                  }
                ]
              },
              {
                "nodeType": "component.object.fieldset",
                "keyName": "share",
                "hocs": "[withTitleAndDescription,withRequest,relation,withRoute,withQuery,routeMiniApp,connectId]",
                "children": [
                  {
                    "nodeType": "component.number.input",
                    "keyName": "showShareBottom",
                    "hocs": "[withTitleAndDescription,withRequest,relation,withRoute,withQuery,routeMiniApp,connectId]"
                  },
                  {
                    "nodeType": "component.number.input",
                    "keyName": "showGoodBottom",
                    "hocs": "[withTitleAndDescription,withRequest,relation,withRoute,withQuery,routeMiniApp,connectId]"
                  }
                ]
              },
              {
                "nodeType": "component.object.fieldset",
                "keyName": "other",
                "hocs": "[withTitleAndDescription,withRequest,relation,withRoute,withQuery,routeMiniApp,connectId]",
                "children": [
                  {
                    "nodeType": "component.string.input",
                    "keyName": "introduction",
                    "hocs": "[withTitleAndDescription,withRequest,relation,withRoute,withQuery,routeMiniApp,connectId]"
                  },
                  {
                    "nodeType": "component.geoPoint.default",
                    "keyName": "position",
                    "hocs": "[withTitleAndDescription,withRequest,relation,withRoute,withQuery,routeMiniApp,connectId]"
                  }
                ]
              }
            ]
          }
        ]
      }
    ]
  },
  "info": {
    "nodeType": "layout.body",
    "children": [
      {
        "nodeType": "component.object.fieldset",
        "keyName": "info",
        "hocs": "[withTitleAndDescription,withRequest,relation,withRoute,withQuery,routeMiniApp,connectId]",
        "children": [
          {
            "nodeType": "layout.block",
            "childrenName": "[name,description]",
            "children": [
              {
                "nodeType": "component.string.input",
                "keyName": "name",
                "hocs": "[withTitleAndDescription,withRequest,relation,withRoute,withQuery,routeMiniApp,connectId]"
              },
              {
                "nodeType": "component.string.input",
                "keyName": "description",
                "hocs": "[withTitleAndDescription,withRequest,relation,withRoute,withQuery,routeMiniApp,connectId]"
              }
            ]
          },
          {
            "nodeType": "component.object.fieldset",
            "keyName": "info",
            "hocs": "[withTitleAndDescription,withRequest,relation,withRoute,withQuery,routeMiniApp,connectId]",
            "children": [
              {
                "nodeType": "component.object.fieldset",
                "keyName": "info2",
                "hocs": "[withTitleAndDescription,withRequest,relation,withRoute,withQuery,routeMiniApp,connectId]",
                "children": [
                  {
                    "nodeType": "component.string.input",
                    "keyName": "name2",
                    "hocs": "[withTitleAndDescription,withRequest,relation,withRoute,withQuery,routeMiniApp,connectId]"
                  }
                ]
              }
            ]
          }
        ]
      }
    ]
  },
  "popup": {
    "nodeType": "layout.body",
    "children": [
      {
        "nodeType": "layout.block",
        "children": [
          {
            "nodeType": "component.array.table",
            "keyName": "popup",
            "hocs": "[withTitleAndDescription,withRequest,relation,miniApp,withRoute,query,routeMiniApp,connectId]",
            "children": [
              {
                "nodeType": "component.string.input",
                "keyName": "name",
                "hocs": "[withTitleAndDescription,withRequest,relation,withRoute,withQuery,routeMiniApp,connectId]"
              }
            ]
          }
        ]
      }
    ]
  },
  "tab": {
    "nodeType": "layout.body",
    "children": [
      {
        "nodeType": "layout.block",
        "children": [
          {
            "nodeType": "component.array.tab",
            "keyName": "tab",
            "hocs": "[withTitleAndDescription,withRequest,relation,withRoute,withQuery,routeMiniApp,connectId]",
            "children": [
              {
                "nodeType": "component.string.input",
                "keyName": "name",
                "hocs": "[withTitleAndDescription,withRequest,relation,withRoute,withQuery,routeMiniApp,connectId]"
              }
            ]
          }
        ]
      }
    ]
  },
  "undefined": {
    "nodeType": "layout.body",
    "children": [
      {
        "nodeType": "layout.block",
        "children": [
          {
            "nodeType": "component.array.table",
            "hocs": "[withTitleAndDescription,withRequest,relation,miniApp,withRoute,query,routeMiniApp,connectId]",
            "children": [
              {
                "nodeType": "component.string.input",
                "keyName": "name",
                "hocs": "[withTitleAndDescription,withRequest,relation,withRoute,withQuery,routeMiniApp,connectId]"
              }
            ]
          }
        ]
      }
    ]
  }
}
`;

exports[`complex schema integration 2 1`] = `
{
  "info": {
    "nodeType": "layout.body",
    "children": [
      {
        "nodeType": "component.object.fieldset",
        "keyName": "info",
        "hocs": "[withTitleAndDescription,withRequest,relation,withRoute,withQuery,routeMiniApp,connectId]",
        "children": [
          {
            "nodeType": "layout.block",
            "childrenName": "[name,test,thumbnail,contactInfo]",
            "children": [
              {
                "nodeType": "layout.block",
                "keyName": "name",
                "childrenName": "[name,nickname]",
                "children": [
                  {
                    "nodeType": "component.string.input",
                    "keyName": "name",
                    "hocs": "[withTitleAndDescription,withRequest,relation,withRoute,withQuery,routeMiniApp,connectId]"
                  },
                  {
                    "nodeType": "component.string.input",
                    "keyName": "nickname",
                    "hocs": "[withTitleAndDescription,withRequest,relation,withRoute,withQuery,routeMiniApp,connectId]"
                  }
                ]
              },
              {
                "nodeType": "component.string.input",
                "keyName": "test",
                "hocs": "[withTitleAndDescription,withRequest,relation,withRoute,withQuery,routeMiniApp,connectId]"
              },
              {
                "nodeType": "component.object.fieldset",
                "keyName": "thumbnail",
                "hocs": "[withTitleAndDescription,withRequest,relation,withRoute,withQuery,routeMiniApp,connectId]",
                "children": [
                  {
                    "nodeType": "component.string.input",
                    "keyName": "imageName",
                    "hocs": "[withTitleAndDescription,withRequest,relation,withRoute,withQuery,routeMiniApp,connectId]"
                  },
                  {
                    "nodeType": "component.file.file",
                    "keyName": "imageSrc",
                    "hocs": "[withTitleAndDescription,withRequest,relation,withRoute,withQuery,routeMiniApp,connectId]"
                  }
                ]
              },
              {
                "nodeType": "component.object.fieldset",
                "keyName": "contactInfo",
                "hocs": "[withTitleAndDescription,withRequest,relation,withRoute,withQuery,routeMiniApp,connectId]",
                "children": [
                  {
                    "nodeType": "component.array.table",
                    "keyName": "phones",
                    "hocs": "[withTitleAndDescription,withRequest,relation,miniApp,withRoute,withQuery,routeMiniApp,connectId]",
                    "children": [
                      {
                        "nodeType": "component.string.select",
                        "keyName": "type",
                        "hocs": "[withTitleAndDescription,withRequest,relation,withRoute,withQuery,routeMiniApp,connectId]"
                      },
                      {
                        "nodeType": "component.string.input",
                        "keyName": "value",
                        "hocs": "[withTitleAndDescription,withRequest,relation,withRoute,withQuery,routeMiniApp,connectId]"
                      }
                    ]
                  },
                  {
                    "nodeType": "component.array.table",
                    "keyName": "addresses",
                    "hocs": "[withTitleAndDescription,withRequest,relation,miniApp,withRoute,withQuery,routeMiniApp,connectId]",
                    "children": [
                      {
                        "nodeType": "component.string.select",
                        "keyName": "type",
                        "hocs": "[withTitleAndDescription,withRequest,relation,withRoute,withQuery,routeMiniApp,connectId]"
                      },
                      {
                        "nodeType": "component.string.input",
                        "keyName": "value",
                        "hocs": "[withTitleAndDescription,withRequest,relation,withRoute,withQuery,routeMiniApp,connectId]"
                      }
                    ]
                  },
                  {
                    "nodeType": "component.array.table",
                    "keyName": "emails",
                    "hocs": "[withTitleAndDescription,withRequest,relation,miniApp,withRoute,withQuery,routeMiniApp,connectId]",
                    "children": [
                      {
                        "nodeType": "component.string.select",
                        "keyName": "type",
                        "hocs": "[withTitleAndDescription,withRequest,relation,withRoute,withQuery,routeMiniApp,connectId]"
                      },
                      {
                        "nodeType": "component.string.input",
                        "keyName": "value",
                        "hocs": "[withTitleAndDescription,withRequest,relation,withRoute,withQuery,routeMiniApp,connectId]"
                      }
                    ]
                  }
                ]
              }
            ]
          }
        ]
      }
    ]
  }
}
`;

exports[`integration with qa-compiler, test object 
  integration 6 1`] = `
{
  "info": {
    "nodeType": "layout.body",
    "children": [
      {
        "nodeType": "component.object.fieldset",
        "keyName": "info",
        "hocs": "[withTitleAndDescription,withRequest,relation,withRoute,withQuery,routeMiniApp,connectId]",
        "children": [
          {
            "nodeType": "component.object.fieldset",
            "keyName": "info1",
            "hocs": "[withTitleAndDescription,withRequest,relation,withRoute,withQuery,routeMiniApp,connectId]",
            "children": [
              {
                "nodeType": "layout.block",
                "childrenName": "[info2]",
                "children": [
                  {
                    "nodeType": "component.object.fieldset",
                    "keyName": "info2",
                    "hocs": "[withTitleAndDescription,withRequest,relation,withRoute,withQuery,routeMiniApp,connectId]",
                    "children": [
                      {
                        "nodeType": "component.string.input",
                        "keyName": "name3",
                        "hocs": "[withTitleAndDescription,withRequest,relation,withRoute,withQuery,routeMiniApp,connectId]"
                      }
                    ]
                  }
                ]
              }
            ]
          }
        ]
      }
    ]
  }
}
`;

exports[`integration with qa-compiler, test object integration 1 1`] = `
{
  "info": {
    "nodeType": "layout.body",
    "children": [
      {
        "nodeType": "component.object.fieldset",
        "keyName": "info",
        "hocs": "[withTitleAndDescription,withRequest,relation,withRoute,withQuery,routeMiniApp,connectId]",
        "children": [
          {
            "nodeType": "layout.block",
            "childrenName": "[name]",
            "children": [
              {
                "nodeType": "component.string.input",
                "keyName": "name",
                "hocs": "[withTitleAndDescription,withRequest,relation,withRoute,withQuery,routeMiniApp,connectId]"
              }
            ]
          }
        ]
      }
    ]
  }
}
`;

exports[`integration with qa-compiler, test object integration 2 1`] = `
{
  "info": {
    "nodeType": "layout.body",
    "children": [
      {
        "nodeType": "component.object.fieldset",
        "keyName": "info",
        "hocs": "[withTitleAndDescription,withRequest,relation,withRoute,withQuery,routeMiniApp,connectId]",
        "children": [
          {
            "nodeType": "layout.block",
            "childrenName": "[name]",
            "children": [
              {
                "nodeType": "component.string.input",
                "keyName": "name",
                "hocs": "[withTitleAndDescription,withRequest,relation,withRoute,withQuery,routeMiniApp,connectId]"
              }
            ]
          },
          {
            "nodeType": "component.string.input",
            "keyName": "nickname",
            "hocs": "[withTitleAndDescription,withRequest,relation,withRoute,withQuery,routeMiniApp,connectId]"
          }
        ]
      }
    ]
  }
}
`;

exports[`integration with qa-compiler, test object integration 3 1`] = `
{
  "info": {
    "nodeType": "layout.body",
    "children": [
      {
        "nodeType": "component.object.fieldset",
        "keyName": "info",
        "hocs": "[withTitleAndDescription,withRequest,relation,withRoute,withQuery,routeMiniApp,connectId]",
        "children": [
          {
            "nodeType": "layout.block",
            "childrenName": "[name,nickname]",
            "children": [
              {
                "nodeType": "component.string.input",
                "keyName": "name",
                "hocs": "[withTitleAndDescription,withRequest,relation,withRoute,withQuery,routeMiniApp,connectId]"
              },
              {
                "nodeType": "component.string.input",
                "keyName": "nickname",
                "hocs": "[withTitleAndDescription,withRequest,relation,withRoute,withQuery,routeMiniApp,connectId]"
              }
            ]
          }
        ]
      }
    ]
  }
}
`;

exports[`integration with qa-compiler, test object integration 4 1`] = `
{
  "info": {
    "nodeType": "layout.body",
    "children": [
      {
        "nodeType": "component.object.fieldset",
        "keyName": "info",
        "hocs": "[withTitleAndDescription,withRequest,relation,withRoute,withQuery,routeMiniApp,connectId]",
        "children": [
          {
            "nodeType": "layout.block",
            "childrenName": "[info1]",
            "children": [
              {
                "nodeType": "component.object.fieldset",
                "keyName": "info1",
                "hocs": "[withTitleAndDescription,withRequest,relation,withRoute,withQuery,routeMiniApp,connectId]",
                "children": [
                  {
                    "nodeType": "component.string.input",
                    "keyName": "name",
                    "hocs": "[withTitleAndDescription,withRequest,relation,withRoute,withQuery,routeMiniApp,connectId]"
                  }
                ]
              }
            ]
          }
        ]
      }
    ]
  }
}
`;

exports[`integration with qa-compiler, test object integration 5 1`] = `
{
  "info": {
    "nodeType": "layout.body",
    "children": [
      {
        "nodeType": "component.object.fieldset",
        "keyName": "info",
        "hocs": "[withTitleAndDescription,withRequest,relation,withRoute,withQuery,routeMiniApp,connectId]",
        "children": [
          {
            "nodeType": "layout.block",
            "childrenName": "[info1]",
            "children": [
              {
                "nodeType": "component.object.fieldset",
                "keyName": "info1",
                "hocs": "[withTitleAndDescription,withRequest,relation,withRoute,withQuery,routeMiniApp,connectId]",
                "children": [
                  {
                    "nodeType": "layout.block",
                    "childrenName": "[name]",
                    "children": [
                      {
                        "nodeType": "component.string.input",
                        "keyName": "name",
                        "hocs": "[withTitleAndDescription,withRequest,relation,withRoute,withQuery,routeMiniApp,connectId]"
                      }
                    ]
                  }
                ]
              }
            ]
          }
        ]
      }
    ]
  }
}
`;

exports[`integration with qa-compiler, test object integration 7 1`] = `
{
  "info": {
    "nodeType": "layout.body",
    "children": [
      {
        "nodeType": "component.object.fieldset",
        "keyName": "info",
        "hocs": "[withTitleAndDescription,withRequest,relation,withRoute,withQuery,routeMiniApp,connectId]",
        "children": [
          {
            "nodeType": "layout.block",
            "childrenName": "[name,description]",
            "children": [
              {
                "nodeType": "component.string.input",
                "keyName": "name",
                "hocs": "[withTitleAndDescription,withRequest,relation,withRoute,withQuery,routeMiniApp,connectId]"
              },
              {
                "nodeType": "component.string.input",
                "keyName": "description",
                "hocs": "[withTitleAndDescription,withRequest,relation,withRoute,withQuery,routeMiniApp,connectId]"
              }
            ]
          },
          {
            "nodeType": "component.object.fieldset",
            "keyName": "info",
            "hocs": "[withTitleAndDescription,withRequest,relation,withRoute,withQuery,routeMiniApp,connectId]",
            "children": [
              {
                "nodeType": "component.object.fieldset",
                "keyName": "info2",
                "hocs": "[withTitleAndDescription,withRequest,relation,withRoute,withQuery,routeMiniApp,connectId]",
                "children": [
                  {
                    "nodeType": "component.string.input",
                    "keyName": "name2",
                    "hocs": "[withTitleAndDescription,withRequest,relation,withRoute,withQuery,routeMiniApp,connectId]"
                  }
                ]
              }
            ]
          }
        ]
      }
    ]
  }
}
`;

exports[`integration with qa-compiler, test object integration 8 1`] = `
{
  "info": {
    "nodeType": "layout.body",
    "children": [
      {
        "nodeType": "component.object.fieldset",
        "keyName": "info",
        "hocs": "[withTitleAndDescription,withRequest,relation,withRoute,withQuery,routeMiniApp,connectId]",
        "children": [
          {
            "nodeType": "layout.block",
            "childrenName": "[name,description]",
            "children": [
              {
                "nodeType": "component.string.input",
                "keyName": "name",
                "hocs": "[withTitleAndDescription,withRequest,relation,withRoute,withQuery,routeMiniApp,connectId]"
              },
              {
                "nodeType": "component.string.input",
                "keyName": "description",
                "hocs": "[withTitleAndDescription,withRequest,relation,withRoute,withQuery,routeMiniApp,connectId]"
              }
            ]
          },
          {
            "nodeType": "component.object.fieldset",
            "keyName": "info",
            "hocs": "[withTitleAndDescription,withRequest,relation,withRoute,withQuery,routeMiniApp,connectId]",
            "children": [
              {
                "nodeType": "component.object.fieldset",
                "keyName": "info2",
                "hocs": "[withTitleAndDescription,withRequest,relation,withRoute,withQuery,routeMiniApp,connectId]",
                "children": [
                  {
                    "nodeType": "layout.block",
                    "childrenName": "[name2]",
                    "children": [
                      {
                        "nodeType": "component.string.input",
                        "keyName": "name2",
                        "hocs": "[withTitleAndDescription,withRequest,relation,withRoute,withQuery,routeMiniApp,connectId]"
                      }
                    ]
                  }
                ]
              }
            ]
          }
        ]
      }
    ]
  },
  "info3": {
    "nodeType": "layout.body",
    "children": [
      {
        "nodeType": "layout.block",
        "children": [
          {
            "nodeType": "component.array.table",
            "keyName": "info3",
            "hocs": "[withTitleAndDescription,withRequest,relation,miniApp,withRoute,query,routeMiniApp,connectId]",
            "children": [
              {
                "nodeType": "layout.block",
                "childrenName": "[name,description]",
                "children": [
                  {
                    "nodeType": "component.string.input",
                    "keyName": "name",
                    "hocs": "[withTitleAndDescription,withRequest,relation,withRoute,withQuery,routeMiniApp,connectId]"
                  },
                  {
                    "nodeType": "component.string.input",
                    "keyName": "description",
                    "hocs": "[withTitleAndDescription,withRequest,relation,withRoute,withQuery,routeMiniApp,connectId]"
                  }
                ]
              },
              {
                "nodeType": "component.object.fieldset",
                "keyName": "info",
                "hocs": "[withTitleAndDescription,withRequest,relation,withRoute,withQuery,routeMiniApp,connectId]",
                "children": [
                  {
                    "nodeType": "component.object.fieldset",
                    "keyName": "info2",
                    "hocs": "[withTitleAndDescription,withRequest,relation,withRoute,withQuery,routeMiniApp,connectId]",
                    "children": [
                      {
                        "nodeType": "component.string.input",
                        "keyName": "name2",
                        "hocs": "[withTitleAndDescription,withRequest,relation,withRoute,withQuery,routeMiniApp,connectId]"
                      }
                    ]
                  }
                ]
              }
            ]
          }
        ]
      }
    ]
  }
}
`;

exports[`integration with qa-compiler, test object integration 9 1`] = `
{
  "info": {
    "nodeType": "layout.body",
    "children": [
      {
        "nodeType": "component.object.fieldset",
        "keyName": "info",
        "hocs": "[withTitleAndDescription,withRequest,relation,withRoute,withQuery,routeMiniApp,connectId]",
        "children": [
          {
            "nodeType": "layout.block",
            "childrenName": "[]",
            "children": [
              {
                "nodeType": "layout.block",
                "childrenName": "[name]",
                "children": [
                  {
                    "nodeType": "component.string.input",
                    "keyName": "name",
                    "hocs": "[withTitleAndDescription,withRequest,relation,withRoute,withQuery,routeMiniApp,connectId]"
                  }
                ]
              }
            ]
          }
        ]
      }
    ]
  }
}
`;
